### PR TITLE
[JS] feat: attachments in final stream chunk

### PR DESCRIPTION
**Pull request by lilyydu**
## Linked issues

closes: #2036

## Details

- add option to add attachments to final stream chunk, by calling `StreamingResponse.setAttachments(attachments)`
- streamer instance is accessed via the `endStreamHandler` (type is `PromptCompletionModelResponseReceivedEvent`)

## Attestation Checklist

- [X] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: js/packages/teams-ai/src/StreamingResponse.ts [hunk 1]
```diff
@@ -140,6 +143,7 @@ export class StreamingResponse {
                 return {
                     type: 'message',
                     text: this._message,
+                    attachments: this._attachments,
```
Comment (rajan-chari) [inline js/packages/teams-ai/src/StreamingResponse.ts:146]: Does it matter that this._attachments can be null?  Or am I reading this wrong?

Comment (lilyydu) [reply js/packages/teams-ai/src/StreamingResponse.ts:146]: nope,  type for attachments can be `Attachment[] | undefined`

File: js/packages/teams-ai/src/StreamingResponse.ts [hunk 2]
```diff
@@ -113,6 +114,22 @@ export class StreamingResponse {
         return this._queueSync!;
     }
 
+    /**
+     * Sets the attachments to attach to the final chunk.
+     * @param attachments List of attachments.
+     */
+    public setAttachments(attachments: Attachment[]): void {
```
Comment (corinagum) [inline js/packages/teams-ai/src/StreamingResponse.ts:121]: This implies that attachments is never undefined, is that correct?

File: js/samples/04.ai-apps/i.teamsChefBot-streaming/src/index.ts [hunk 3]
```diff
@@ -104,11 +106,34 @@ const prompts = new PromptManager({
     promptsFolder: path.join(__dirname, '../src/prompts')
 });
 
+const endStreamHandler: PromptCompletionModelResponseReceivedEvent = (ctx, memory, response, streamer) => {
+    // Ignore events for other contexts
+    if (!streamer) {
```
Comment (corinagum) [inline js/samples/04.ai-apps/i.teamsChefBot-streaming/src/index.ts:111]: Are there currently other events that might be caught here or is this a guard for the future?

**Review by corinagum — CHANGES_REQUESTED** [review]
Just need to remove the yarn.lock, otherwise LGTM. I left some questions as well.

File: js/packages/teams-ai/src/StreamingResponse.ts (hunk 2 above)
Comment (lilyydu) [reply js/packages/teams-ai/src/StreamingResponse.ts:121]: if they are sending attachments, then yes

File: js/samples/04.ai-apps/i.teamsChefBot-streaming/src/index.ts (hunk 3 above)
Comment (lilyydu) [reply js/samples/04.ai-apps/i.teamsChefBot-streaming/src/index.ts:111]: for the future

**Review by corinagum — APPROVED** [review]

**Review by rajan-chari — APPROVED** [review]
Thanks Lily

**Pull request merged**